src/Swift.Bindings/tests/CryptoKit/Foundation.Source.cs

```diff
@@ -0,0 +1,274 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Reflection;
+
+namespace Test
+{
+    /// <summary>
+    /// Represents a Swift type in C#.
+    /// </summary>
+    public unsafe interface ISwiftObject
+    {
+        public static abstract void* Metadata { get; }
+    }
+
+    // <summary>
+    // Represents Swift UnsafePointer in C#.
+    // </summary>
+    public readonly unsafe struct UnsafePointer<T> where T : unmanaged
+    {
+        private readonly T* _pointee;
+        public UnsafePointer(T* pointee)
+        {
+            this._pointee = pointee;
+        }
+
+        public T* Pointee => _pointee;
+
+        public static implicit operator T*(UnsafePointer<T> pointer) => pointer.Pointee;
+
+        public static implicit operator UnsafePointer<T>(T* pointee) => new(pointee);
+    }
+
+    // <summary>
+    // Represents Swift UnsafeMutablePointer in C#.
+    // </summary>
+    public readonly unsafe struct UnsafeMutablePointer<T> where T : unmanaged
```

Comment:
We don't need to correct this now, but this code will cause memory leaks without it honoring the Swift memory model. Please open an issue for this.

Reply:
Added as a task `Review memory management of the projected pointer types` in https://github.com/dotnet/runtime/issues/95633.